src/basic/io-util.h

```diff
@@ -44,3 +44,5 @@ static inline bool FILE_SIZE_VALID_OR_INFINITY(uint64_t l) {
         return FILE_SIZE_VALID(l);
 
 }
+
+void report_errno_and_exit(int errno_fd, int error);
```

Comment:
Please also mark prototype as noreturn

Reply:
BTW, shouldn't this live in process-util?

Reply:
> BTW, shouldn't this live in process-util?

Indeed, it seems a better fit.